### PR TITLE
[Bugfix] Emit a valid media type from encode_{audio,image,video}_url

### PR DESCRIPTION
**Pull request by vineethsaivs**
## Purpose

`encode_audio_url`, `encode_image_url` and `encode_video_url` in `vllm/multimodal/utils.py` build a data URL and resolve its media type with:

```python
mimetype = mimetypes.types_map.get("." + format.lower(), "image")  # or "audio"/"video"
return f"data:{mimetype};base64,{...}"
```

When the format's extension is not registered in `mimetypes.types_map`, the fallback is a bare top-level type (`"image"`, `"audio"`, `"video"`). A media type with no subtype is not well-formed, so formats missing from the table produce an invalid data URL such as `data:image;base64,...`. This is reachable with formats Pillow/soundfile can encode but that Python's `mimetypes` does not register, for example `TGA`/`PCX` images or `CAF`/`W64` audio.

This changes the fallback to `"<kind>/<format>"` (e.g. `image/tga`), so the media type is always a well-formed `type/subtype`. Registered formats are unchanged (e.g. `PNG` still resolves to `image/png`).

This only affects the media-type string of the generated data URL (a request-construction helper); it does not touch model output, accuracy, or serving numerics, so no model-evaluation results apply.

**Not a duplicate:** I searched open PRs for `encode_image_url`/`encode_audio_url`, `mimetype`, and `multimodal/utils.py`; the open multimodal PRs (#43638, #46947, #44474, #29034, ...) touch examples, spec-decode embeddings, and video URL loading, none change the `encode_*_url` media-type fallback. No open issue tracks this.

## Test Plan

Added `tests/multimodal/test_utils.py::test_encode_image_url_uncommon_format_has_valid_mimetype`, which encodes a Pillow image with `format="TGA"` (`.tga` is not in `mimetypes.types_map`) and asserts the media type is `image/tga`:

```bash
pytest tests/multimodal/test_utils.py::test_encode_image_url_uncommon_format_has_valid_mimetype -q
```

## Test Result

A full CUDA build was not available on my machine, so I validated the exact fixed function bodies on CPU by extracting `encode_{audio,image,video}_url` from the module and executing them with the base64 encoders stubbed (the media-type logic is independent of encoding). The added test runs in CI.

- Before, uncommon formats produced bare types: `encode_image_url(img, format="TGA")` -> `data:image;base64,...`, `encode_video_url(frames, format="MKV")` -> `data:video;base64,...`.
- After: `data:image/tga;base64,...`, `data:audio/caf;base64,...`, `data:video/mkv;base64,...` (all well-formed `type/subtype`).
- Registered formats unchanged: `PNG` -> `image/png`, `WAV` -> `audio/x-wav`.

Ran `ruff check` and `ruff format --check` on both changed files: clean.

---

**AI assistance disclosure:** AI assistance was used to help identify and prepare this change, per the repository's contribution policy. The fix updates a one-line fallback in three sibling helpers plus a regression test, and I have reviewed every changed line.

### CODE REVIEW

**Comment by vineethsaivs** [comment]
Gentle ping @DarkLight1337 / @ywang96 (CODEOWNERs for `vllm/multimodal`).

CI is currently blocked at `pre-run-check`: *"PR must have the 'verified', 'ready', or 'ready-run-all-tests' label ... or the author must have at least 4 merged PRs (found 0)"*. As a first-time contributor I cannot add that label myself, so the test suite has never run on this PR. Could you add `ready` (or review) when you have a moment?

Recap: `encode_audio_url` / `encode_image_url` / `encode_video_url` build a data URL whose media type is the bare format string, so the emitted prefix is e.g. `data:jpeg;base64,` rather than `data:image/jpeg;base64,`, which is not a valid media type. The fix prefixes the correct top-level type per helper, with a regression test in `tests/multimodal/test_utils.py`.

File: tests/multimodal/test_utils.py [hunk 1]
```diff
@@ -199,6 +203,18 @@ def test_group_and_batch_mm_items_split_by_fieldset():
     assert [num_items for num_items, _ in res] == [2, 1, 1, 1]
 
 
+def test_encode_image_url_uncommon_format_has_valid_mimetype():
+    """A format missing from mimetypes.types_map must still produce a
+    well-formed type/subtype media type, not a bare 'image'."""
+    from PIL import Image
```
Comment (DarkLight1337) [inline tests/multimodal/test_utils.py]: Please put the imports at the top

Comment (vineethsaivs) [reply tests/multimodal/test_utils.py]: Done in `086ea16`, moved to the third-party block at the top:

```python
import pytest
import torch
from PIL import Image

from vllm.multimodal.inputs import (
```

Matches `tests/multimodal/test_image.py`, which imports PIL the same way.

Thanks for the label as well, the suite had never run on this PR before because of the first-contributor `pre-run-check` gate.

**Review by DarkLight1337 — APPROVED** [review]
Thanks

**Comment by vineethsaivs** [comment]
Thanks for the review and the approval.

One thing that needs a maintainer hand: the red `pre-run-check` is stale rather than a real failure. It ran at 2026-07-26T16:22:48Z against my import-move commit and reported `found 0` merged PRs with no ready label, and the `ready` label was only applied at 2026-07-27T03:57:55Z, about eleven hours later, when the approval landed. The check reads the labels at run time, so a re-run of that job should pass now without any change to the branch.

Happy to push an empty commit to retrigger it instead if you would rather not re-run by hand, though that would recompute the other checks too.

**Comment by vineethsaivs** [comment]
Rebased onto main; mergeable again. The three edited lines in `vllm/multimodal/utils.py` are unchanged upstream, so the conflict was entirely in `tests/multimodal/test_utils.py`, where this test and the new `test_group_and_batch_mm_items_splits_shared_data_by_dtype` had been inserted at the same anchor. Both are kept, with this one after it. @DarkLight1337's import fix is carried over unchanged.
